File: Python scripts/backup_folder.py
import zipfile
import os

# Path to the folder you want to back up
# On Windows, for example: C:\Users\Baha\Desktop\Developpement full stack\Lab 6\logs
folder_to_backup = r"C:\Users\Baha\Desktop\Developpement full stack\Lab 6\logs"

# Output ZIP file
# On Windows, you can save it in the same folder or anywhere else
output_zip = r"C:\Users\Baha\Desktop\Developpement full stack\Lab 6\backup.zip"

# Create the ZIP archive
with zipfile.ZipFile(output_zip, 'w', zipfile.ZIP_DEFLATED) as zipf:
    for root, dirs, files in os.walk(folder_to_backup):
        for file in files:
            file_path = os.path.join(root, file)
            # Keep folder structure relative to the folder_to_backup
            arcname = os.path.relpath(file_path, folder_to_backup)
            zipf.write(file_path, arcname)

print(f"Backup created at {output_zip}")

File: backup_folder/backup_folder.go
package main

import (
	"archive/zip"
	"fmt"
	"io"
	"os"
	"path/filepath"
)

func main() {
	// Folder to backup
	folder := `C:\Users\Baha\Desktop\Developpement full stack\Lab 6\logs`
	// Output ZIP file
	zipFile := `C:\Users\Baha\Desktop\Developpement full stack\Lab 6\backupp.zip`

	zipf, err := os.Create(zipFile)
	if err != nil {
		fmt.Println("Error creating ZIP:", err)
		return
	}
	defer zipf.Close()

	archive := zip.NewWriter(zipf)
	defer archive.Close()

	err = filepath.Walk(folder, func(path string, info os.FileInfo, err error) error {
		if err != nil {
			fmt.Println("Error walking path:", err)
			return err
		}
		if info.IsDir() {
			return nil // skip directories
		}

		// Relative path inside the zip
		relPath, err := filepath.Rel(folder, path)
		if err != nil {
			return err
		}

		file, err := os.Open(path)
		if err != nil {
			return err
		}
		defer file.Close()

		w, err := archive.Create(relPath)
		if err != nil {
			return err
		}

		_, err = io.Copy(w, file)
		return err
	})

	if err != nil {
		fmt.Println("Error:", err)
		return
	}

	fmt.Println("Backup created at", zipFile)
}
